search/query.go

```diff
@@ -44,60 +47,150 @@ type PostSearchResult struct {
 	Post any        `json:"post"`
 }
 
-func doSearchPosts(ctx context.Context, escli *es.Client, q string, offset int, size int) (*EsSearchResponse, error) {
+func checkParams(offset, size int) error {
+	if offset+size > 5000 || size > 1000 || offset > 1000 || offset < 0 || size < 0 {
+		return fmt.Errorf("disallowed size/offset parameters")
+	}
+	return nil
+}
+
+func DoSearchPosts(ctx context.Context, dir identity.Directory, escli *es.Client, index, q string, offset, size int) (*EsSearchResponse, error) {
+	if err := checkParams(offset, size); err != nil {
+		return nil, err
+	}
+	queryStr, filters := ParseQuery(ctx, dir, q)
+	basic := map[string]interface{}{
+		"simple_query_string": map[string]interface{}{
+			"query":            queryStr,
+			"fields":           []string{"everything"},
```

Comment:
I wonder how this is going to impact searching for words that people have in their handles

Reply:
For the post schema, we don't include handles in the document, or in "everything", so should not impact.